### PR TITLE
fix incorrect and missing defaults for content settings

### PR DESCRIPTION
**Pull request by bridiver**
Resolves https://github.com/brave/brave-browser/issues/23031

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by goodov** [comment]
@boocmp fyi

File: browser/content_settings/brave_content_settings_registry_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+  SCOPED_TRACE("Content setting: BRAVE_SPEEDREADER");
+  info = registry()->Get(ContentSettingsType::BRAVE_SPEEDREADER);
+  EXPECT_EQ(CONTENT_SETTING_ALLOW, info->GetInitialDefaultSetting());
+
+  SCOPED_TRACE("Content setting: BRAVE_ETHEREUM");
+  info = registry()->Get(ContentSettingsType::BRAVE_ETHEREUM);
+  EXPECT_EQ(CONTENT_SETTING_ASK, info->GetInitialDefaultSetting());
+
+  SCOPED_TRACE("Content setting: BRAVE_SOLANA");
```
Comment (goodov) [inline browser/content_settings/brave_content_settings_registry_unittest.cc:152]: there's a lot of `SCOPED_TRACE` in the same scope, they all will be printed on error.

File: components/brave_shields/browser/cookie_pref_service.h [hunk 2]
```diff
@@ -29,6 +31,9 @@ class CookiePrefService : public KeyedService,
   CookiePrefService& operator=(const CookiePrefService&) = delete;
   ~CookiePrefService() override;
 
+  static ControlType CookieControlsModeToControlType(
```
Comment (iefremov) [inline components/brave_shields/browser/cookie_pref_service.h:34]: why is it here now? can't find where it is used outside of .cc

File: components/content_settings/core/browser/brave_content_settings_default_provider.cc [hunk 3]
```diff
@@ -26,9 +26,6 @@ BraveDefaultProvider::~BraveDefaultProvider() {}
 std::unique_ptr<RuleIterator> BraveDefaultProvider::GetRuleIterator(
     ContentSettingsType content_type,
     bool off_the_record) const {
-  if (content_settings::IsShieldsContentSettingsType(content_type))
```
Comment (iefremov) [inline components/content_settings/core/browser/brave_content_settings_default_provider.cc:29]: why this change is needed?

File: chromium_src/components/content_settings/core/common/content_settings.cc [hunk 4]
```diff
@@ -100,9 +84,19 @@ void RendererContentSettingRules::FilterRulesByOutermostMainFrameURL(
   RendererContentSettingRules_ChromiumImpl::FilterRulesByOutermostMainFrameURL(
       outermost_main_frame_url);
   FilterRulesForType(autoplay_rules, outermost_main_frame_url);
-  FilterRulesForType(fingerprinting_rules, outermost_main_frame_url);
   FilterRulesForType(brave_shields_rules, outermost_main_frame_url);
-  FilterRulesForType(cosmetic_filtering_rules, outermost_main_frame_url);
+  // FilterRulesForType has a DHCECK on the size and these fail (for now)
```
Comment (iefremov) [inline chromium_src/components/content_settings/core/common/content_settings.cc:88]: typo DCHECK

File: components/content_settings/core/browser/brave_content_settings_default_provider.cc (hunk 3 above)
Comment (iefremov) [reply components/content_settings/core/browser/brave_content_settings_default_provider.cc:29]: i think i've got the idea, but if we remove this check then the whole class seems useless

File: browser/content_settings/brave_content_settings_registry_unittest.cc [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+  {
+    SCOPED_TRACE("Content setting: BRAVE_SOLANA");
+    info = registry()->Get(ContentSettingsType::BRAVE_SOLANA);
+    EXPECT_EQ(CONTENT_SETTING_ASK, info->GetInitialDefaultSetting());
+  }
+
+  // TODO(bridiver) - DEFAULT is not a valid setting, but leaving it for
+  // now because this requires more extensive changes to fix correctly
+  // {
+  //   SCOPED_TRACE("Content setting: BRAVE_FINGERPRINTING_V2");
```
Comment (iefremov) [inline browser/content_settings/brave_content_settings_registry_unittest.cc:167]: i'd rather keep checks that assert "not valid" things then keep something commented and not even compiled

**Review by iefremov — APPROVED** [review]

File: browser/content_settings/brave_content_settings_registry_unittest.cc (hunk 5 above)
Comment (bridiver) [reply browser/content_settings/brave_content_settings_registry_unittest.cc:167]: uncommenting these causes a crash because GetInitialDefaultSetting DCHECKs for invalid values

File: components/content_settings/core/browser/brave_content_settings_default_provider.cc (hunk 3 above)
Comment (bridiver) [reply components/content_settings/core/browser/brave_content_settings_default_provider.cc:29]: indeed, I forgot to remove this once I verified that everything was working correctly

**Comment by bridiver** [comment]
unrelated test failures on macos
